python/nav/models/manage.py

```diff
@@ -368,7 +377,7 @@ def get_preferred_snmp_management_profile(self, writeable=None):
             )
         profiles = sorted(
             self.profiles.filter(query),
-            key=lambda p: str(p.configuration.get('version') or 0),
+            key=lambda p: p.snmp_version or 0,
```

Comment:
Probably because it was written before the `snmp_version` property was added to the `ManagementProfile` class.

Either that, or it could have been a protective measure against faulty profiles that are somehow missing the version option.  But I think I'd rather have that blow up in our face than go away silently...

Reply:
And, as it the case here, SNMPv3 profile do not actually have the `version` option, so the `snmp_version` property wins the battle here...